### ISSUE TITLE
Avoid raising S310 on safe f-strings

### ISSUE THREAD

**Issue by dscorbett** [astral-sh/ruff#12245]
Given this input file (s310_fstring.py):
```python
from urllib.request import urlopen
foo = "foo"
urlopen(f"https://www.example.com/{foo}")
```
Ruff reports a violation of rule S310.
```console
$ ruff check --select S310 --output-format concise s310_fstring.py
s310_fstring.py:3:1: S310 Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.
Found 1 error.
```
Even though the argument is an f-string, Ruff should be able to detect that it begins with a permitted scheme. Compare #8040.

**Comment by dhruvmanila** [comment]
Yeah, I think this makes sense. We can check for the first f-string literal part and verify if it begins with `http` or `https` similar to string literal.
